src/main/java/nl/ordina/jobcrawler/scrapers/YachtVacancyScraper.java

```diff
@@ -1,149 +1,149 @@
+                String vacancyURL = (String) vacancyData.get("detailUrl");
+                String title = (String) vacancyData.get("title");
+                String hours = (String) vacancyMetaData.get("hours");
+                String location = (String) vacancyMetaData.get("location");
+                String vacancyNumber = (String) vacancyData.get("vacancyNumber");
+                String postingDate = (String) vacancyData.get("date");
+                vacancyURL =  vacancyURL.contains("?") ? vacancyURL.split("\\?")[0] : vacancyURL;
                 vacancyURLs.add(
                         VacancyURLs.builder()
-                        .url(vacancyURL)
-                        .hours(hour)
+                        .url(VACANCY_URL_PREFIX + vacancyURL)
+                        .title(title)
+                        .hours(hours)
+                        .location(location)
+                        .vacancyNumber(vacancyNumber)
+                        .postingDate(postingDate)
                         .build()
                 );
             }
         }
         return vacancyURLs;
     }
 
+    /**
+     * This method does a get request to Yacht to retrieve the vacancies from a specific page.
+     * @param pageNumber Pagenumber of which the vacancy data should be retrieved
+     * @return json response from the get request
+     */
+    private YachtVacancyResponse scrapeVacancies(int pageNumber) {
+        RestTemplate restTemplate = new RestTemplate();
+        MappingJackson2HttpMessageConverter mappingJackson2HttpMessageConverter = new MappingJackson2HttpMessageConverter();
+        mappingJackson2HttpMessageConverter.setSupportedMediaTypes(Arrays.asList(MediaType.APPLICATION_JSON, MediaType.APPLICATION_OCTET_STREAM));
+        restTemplate.getMessageConverters().add(mappingJackson2HttpMessageConverter);
+
+        ResponseEntity<YachtVacancyResponse> response
+                = restTemplate.getForEntity(getSEARCH_URL() + "&pagina=" + pageNumber, YachtVacancyResponse.class);
+
+        return response.getBody();
+    }
+
     @Override
     protected int getTotalNumberOfPages(Document doc) {
-        // In case there are a lot of pages, we can easily detect the last page by getting the id from the double arrow that goes to the last page.
-        Elements pagesElement = doc.select("li.last.last-short a");
-        if(pagesElement.isEmpty()) {
-            // If only a few pages are available the double arrow is disabled and does not contain an ID. Now we need to request the latest page via li.pages and select the last entry.
-            Elements fewPagesElement = doc.select("li.pages ul li a");
-            if(fewPagesElement.isEmpty())
-                return 1;
-
-            return Integer.parseInt(fewPagesElement.last().attr("id"));
-        }
-        else {
-            return Integer.parseInt(pagesElement.attr("id"));
-        }
+        return 1;
```

Comment:
Please add the reason why this always returns 1

Reply:
Done 27fd1b8